src/boot/efi/boot.c

```diff
@@ -1926,14 +1926,14 @@ static bool is_sd_boot(EFI_FILE *root_dir, const char16_t *loader_path) {
                         /* profile= */ UINT_MAX,
                         /* validate_base= */ 0,
                         &vector);
-        if (vector.memory_size != sizeof(SD_MAGIC))
+        if (vector.memory_size != sizeof(SD_MAGIC) - 1)
                 return false;
 
         err = file_handle_read(handle, vector.file_offset, vector.file_size, &content, &read);
         if (err != EFI_SUCCESS || vector.file_size != read)
                 return false;
 
-        return memcmp(content, SD_MAGIC, sizeof(SD_MAGIC)) == 0;
+        return memcmp(content, SD_MAGIC, sizeof(SD_MAGIC) - 1) == 0;
```

Comment:
hmm, but means we are not going to recognize old versions of sd-boot anymore.

i think there's nothing wrong with appending a NUL byte to the .sdmagic PE section given that we defined it on our own. So I would just continue doing that, for compat.

Reply:
hmm, maybe it would suffice to suffix SD_MAGIC with an explicit `"\0"`, but no idea how the assembler is going to like that and if it does the right thing.

Reply:
SD_MAGIC is defined as `"#### LoaderInfo: systemd-boot " GIT_VERSION " ####"`, so I won't match older versions anyway, so this compat is not useful?

Reply:
oh, right. ignore me then.

but maybe just use STRLEN() then

Reply:
STRLEN doesn't work here because it's efi code. I tried already.

Reply:
maybe move the STRLEN() definitoin into src/fundamental?

Reply:
wait, STRLEN() already is already in src/fundamental/macro-fundamental.h. You can just use it.

Reply:
Oh, I think I wrote `STRLEN`, then I changed that to strlen, which didn't compile. `STRLEN` compiles and is pretty much the same as the open-coded `sizeof() - 1`. I changed it back to `STRLEN`.